tests/integration/local/test_shared_lib_include.py

```diff
@@ -0,0 +1,136 @@
+
+        # The user can include directly the header related to the library
+        # or choose to declare functions or objects later
+        header_path = (
+        	"tests/integration/local/"
+        	"test_shared_libraries/a.h"
+        )
+        PyRDF.include_headers(header_path)
+
+        # Creates an RDataFrame with 5 integers [0...4]
+        rdf = PyRDF.RDataFrame(5)
+
+        # This filters out all numbers less than 3
+        filter1 = rdf.Filter("tdfentry_ < 3")
+
+        # This defines a new variable x with all entries squared
+        # then filters out all x values less than 3
+        filter2 = rdf.Define("x","f(tdfentry_)").Filter("x < 3")
+
+        count1 = filter1.Count().GetValue()
+        count2 = filter2.Count().GetValue()
+
+        # The final answer should be the number of integers
+        # less than 5, which is 3, and the number of squared integers less
+        # than 5, which is 2.
+        self.assertEqual(count1, 3)
+        self.assertEqual(count2, 2)
+
+        # Remove library at the end so it doesn't get uploaded with the project
+        os.remove(so_path)
+
+    def test_includes_shared_lib_with_pcm(self):
+        """
+        Check that a pcm file is correctly included with its shared library.
+        """
+        # Get path of current working directory
+        root_dir = os.getcwd()
+        # Change working directory to create shared library
+        wd_path = "tests/integration/local/test_shared_libraries/"
+        os.chdir(wd_path)
+
+        h_path = "myclass.h"
+        cxx_path = "myclass.cxx"
+        library_path = "libmyclass.so"
+        pcm_path = "myclass_rdict.pcm"
+
+        # Create the pcm file from the header
+        pcm_code = "genreflex {h} -o {cxx}".format(h = h_path, cxx = cxx_path)
+        subprocess.call(pcm_code, shell = True)
+
+        # Create the shared library in the same folder as the pcm
+        library_code = (
+            "g++ -shared -fPIC "
+            "`root-config --cflags --libs` "
+            "{cxx} "
+            "-o {lib}"
+        ).format(cxx = cxx_path, lib = library_path)
+        subprocess.call(library_code, shell = True)
+
+        # Change back to initial working directory
+        os.chdir(root_dir)
+
+        # Disable ROOT autoparsing, this way if the pcm is not present it will
+        # trigger an error
+        apState = ROOT.gInterpreter.SetClassAutoparsing(0)
+
+        # Include the shared library. ROOT will know the class declared inside,
+        # but not its data members (those are in the pcm file).
+        PyRDF.include_shared_libraries(wd_path+library_path)
+
+        # Ask ROOT to return an instance of the class in the shared library
+        c = ROOT.TClass.GetClass("myclass")
+
+        # Get the name of the data member from the class through ROOT
+        # if succesfull, it means that ROOT read it from the pcm file.
+        self.assertEqual(c.GetRealData("pi").GetName(), "pi")
```

Comment:
be careful with equality of floats: https://stackoverflow.com/questions/8929005/unittest-sometimes-fails-because-floating-point-imprecision